### DISCUSSION TITLE
How to open DateTimePicker onClick of a button?

### DISCUSSION THREAD

**Discussion by souhagaa** [mantinedev/mantine#6208] [Q&A] [answered]
Is it possible to make DateTimePicker open on the click of an icon? I am using leftSection prop to add a calendar icon to the input but I'm not sure how to make the picker open up on click of the icon 
I only managed to set the focus using a ref on the picker and `onClick={() => inputRef.current?.inputElement?.focus()}` on the icon but no idea how to actually open the popover onClick
Thank you in advance any help is appreciated 😄

**Comment by rtivital** [comment]
onClick={() => inputRef.current?.inputElement?.click()}

**Reply by souhagaa** [reply]
@rtivital thanks for your answer, I tried that but it doesn't work I get a TS error `Property 'inputElement' does not exist on type 'HTMLButtonElement'` and nothing happens when I click .. Here is a [codesandbox](https://codesandbox.io/p/sandbox/mantine-react-template-forked-jlmdwv?file=%2Fsrc%2FApp.tsx%3A18%2C20&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvzn4nl000063b6impbh7tw9%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvzn4nl000023b6ircrd9vij%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvzn4nl000033b6iqs70xq6i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvzn4nl000053b6iq49manaf%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvzn4nl000023b6ircrd9vij%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvzn4nl000013b6imp9i4fps%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvzn59lg003w3b6iu6c53ii9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A18%252C%2522startColumn%2522%253A20%252C%2522endLineNumber%2522%253A18%252C%2522endColumn%2522%253A20%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvzn4nl000023b6ircrd9vij%2522%252C%2522activeTabId%2522%253A%2522clvzn59lg003w3b6iu6c53ii9%2522%257D%252C%2522clvzn4nl000053b6iq49manaf%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvzn4nl000043b6ig6o9cdro%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvzn4nl000053b6iq49manaf%2522%252C%2522activeTabId%2522%253A%2522clvzn4nl000043b6ig6o9cdro%2522%257D%252C%2522clvzn4nl000033b6iqs70xq6i%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvzn4nl000033b6iqs70xq6i%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) imitating the code I have 
```
  const inputEl = useRef<HTMLButtonElement>(null);

 const leftSection = useMemo(
    () => (
      <ActionIcon
        onClick={() => inputEl.current?.inputElement?.click()}
      >
        <Icon icon="calendar" />
      </ActionIcon>
    ),
    [],
  );

<DateTimePicker
ref={inputEl} 
leftSection={leftSection}
.... />

```

**Reply by rtivital** [reply, marked as answer]
https://codesandbox.io/p/sandbox/mantine-react-template-forked-zm4lz6?file=%2Fsrc%2FApp.tsx%3A5%2C10

**Reply by souhagaa** [reply]
Thank you so much for the guidance 🙏
